ConvertEventsToMDTest fails sometimes

It failed like this last night for the master-clean win7 build:
```
03:02:44 Error writing to log file path to default location: "C:\Users\builder\AppData\Local\Temp\mantid.log". Will use a system temp path instead: "C:\Users\builder\AppData\Local\Temp\mantid.log"
03:02:44 Running 1 testConvertToMD-[Notice] ConvertToMD started
03:02:44 MDWSTransform-[Notice] Can not obtain transformation matrix from the input workspace: testEvWS as no oriented lattice has been defined. 
03:02:44 ConvertToMD-[Notice] ConvertToMD successful, Duration 0.07 seconds
03:02:44 
03:02:44 In ConvertEventsToMDTest::testEventWS:
03:02:44 C:/Jenkins/workspace/master_clean-win7/Framework/MDAlgorithms/test/ConvertEventsToMDTest.h:61: Error: Expected (900 == ws->getNPoints()), found (900 != 899)
03:02:44 Failed 1 of 1 test
03:02:44 Success rate: 0%
03:02:44 Error in logging framework: Cannot open fileError in logging framework: Cannot open file
```

It had been seen failing on win7 debug sometimes too (#14505). I only remember failures on win7, but this may be a false impression as it doesn't seem to be easy to replicate.

--- a/Framework/MDAlgorithms/test/ConvertEventsToMDTest.h
+++ b/Framework/MDAlgorithms/test/ConvertEventsToMDTest.h
@@ -46,7 +46,7 @@ public:
 
     pAlg->setRethrows(false);
     pAlg->execute();
-    TSM_ASSERT("Shoud finish succesfully", pAlg->isExecuted());
+    TSM_ASSERT("Should finish succesfully", pAlg->isExecuted());
     Mantid::API::Workspace_sptr spws;
     TS_ASSERT_THROWS_NOTHING(
         spws = AnalysisDataService::Instance().retrieve("testMDEvWorkspace"));
@@ -55,7 +55,7 @@ public:
     boost::shared_ptr<DataObjects::MDEventWorkspace<DataObjects::MDEvent<3>, 3>>
         ws = boost::dynamic_pointer_cast<
             DataObjects::MDEventWorkspace<DataObjects::MDEvent<3>, 3>>(spws);
-    TSM_ASSERT("It shoudl be 3D MD workspace", ws.get());
+    TSM_ASSERT("It should be 3D MD workspace", ws.get());
 
     if (ws.get()) {
       TS_ASSERT_EQUALS(900, ws->getNPoints());

--- a/Framework/TestHelpers/src/WorkspaceCreationHelper.cpp
+++ b/Framework/TestHelpers/src/WorkspaceCreationHelper.cpp
@@ -68,9 +68,14 @@ Workspace2D_sptr Create1DWorkspaceRand(int size) {
   MantidVecPtr x1, y1, e1;
   x1.access().resize(size, 1);
   y1.access().resize(size);
-  std::generate(y1.access().begin(), y1.access().end(), rand);
+
+  MersenneTwister randomGen(DateAndTime::getCurrentTime().nanoseconds(), 0,
+                            std::numeric_limits<int>::max());
+  auto randFunc = [&randomGen] { return randomGen.nextValue(); };
+
+  std::generate(y1.access().begin(), y1.access().end(), randFunc);
   e1.access().resize(size);
-  std::generate(e1.access().begin(), e1.access().end(), rand);
+  std::generate(e1.access().begin(), e1.access().end(), randFunc);
   Workspace2D_sptr retVal(new Workspace2D);
   retVal->initialize(1, size, size);
   retVal->setX(0, x1);
@@ -760,13 +765,16 @@ EventWorkspace_sptr CreateRandomEventWorkspace(size_t numbins, size_t numpixels,
   }
   pAxis0->setUnit("TOF");
 
+  MersenneTwister randomGen(DateAndTime::getCurrentTime().nanoseconds(), 0,
+                            std::numeric_limits<int>::max());
   // Make up some data for each pixels
   for (size_t i = 0; i < numpixels; i++) {
     // Create one event for each bin
     EventList &events = retVal->getEventList(static_cast<detid_t>(i));
     for (std::size_t ie = 0; ie < numbins; ie++) {
       // Create a list of events, randomize
-      events += TofEvent(std::rand(), std::rand());
+      events += TofEvent(static_cast<double>(randomGen.nextValue()),
+                         static_cast<int64_t>(randomGen.nextValue()));
     }
     events.addDetectorID(detid_t(i));
   }

